Make gem-man actually work with recent RubyGems versions

rymai opened the pull request:
Hi Chris,

I was so sad that gem-man didn't find any manuals with current the RubyGems that I've made some changes to make it work (keeping the backward compatibility though). I've also made some few changes to update the .gemspec, like replacing http with https in the GitHub's URL! ;)

So now, I'm able to enjoy [Guard's man](https://github.com/guard/guard/blob/master/man/guard.md)!!

Commit:
Use new RubyGems methods instead of the deprecated ones (keep backward compatibility with Rubygems < 1.8).

Commit:
Define a new man_dir method to access the man/ directory (again, use new RubyGems method with backward compatibility).

Commit:
Update .gemspec and replace double quotes with single quote (style matters!!).

Commit:
Replace size.zero? by empty?

davetron5000 commented:
Can you comment which version of RubyGems breaks gem-man (and thus requires this patch)?  I'm having a problem that basically no man pages can be found and want to know if this would fix it, or if there's another problem.  I'm on RubyGems 1.6.2

rymai commented:
I'll have to digg RubyGems former code deeper, but I think it's before 1.7 for sure... And maybe even before 1.6.2...

Sent from my iPhone

On 13 août 2011, at 18:15, davetron5000 reply@reply.github.com wrote:

> Can you comment which version of RubyGems breaks gem-man (and thus requires this patch)?  I'm having a problem that basically no man pages can be found and want to know if this would fix it, or if there's another problem.  I'm on RubyGems 1.6.2
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/defunkt/gem-man/pull/7#issuecomment-1797862

rymai commented:
Ok, I checked RubyGems code and it seems that `gem-man` broke with the v1.6.0 of RubyGems.

rymai commented:
Just for a quick explanation on why it was failing:

RubyGems since 1.6.0 caches gems' .gemspec in a folder but removes the `files` list in it (for faster indexing purpose it seems).

The problem was that gem-man was using these cached specifications by accessing the `files` attributes in them to check if man pages was present in the gem. Since this attribute was stripped in these cached gemspec, no manual pages was found! I believe though that it could possible to show a manual for a gem that has been installed with a version of RubyGems prior to 1.6.0, hence issue #5.

I've now fixed it by checking the actual files of the gems to see if manual pages are present. And I also replaced deprecated  methods at the same time.
